bin/setup

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env sh
+
+# Set up Rails app. Run this script immediately after cloning the codebase.
+
+# Exit if any subcommand fails
+set -e
+
+# Set up ruby version file
+if [ ! -f .ruby-version ]; then
+  cp .ruby-version.example .ruby-version
+fi
+
+# Set up configurable environment variables
+if [ ! -f .env ]; then
+  cp config/env .env
+fi
+
+# Set up Ruby dependencies via Bundler
+gem install bundler --conservative
+bundle check || bundle install --path=vendor/bundle
```

Comment:
@johnpaulashenfelter Could I kindly ask your rationale for vendoring gems in development? There are good reasons for vendoring gems for production deployments, but in my experience, vendoring in development has given my teams more trouble than the default behavior installing gems on the system.

Right now, I'm working on reducing the docker development image build time. Vendored gems make the docker context directory size large (+1.1 G in vendor/), causing longer build times. Additionally, as an individual developer, rather than being able to inherit most or all of my gems from an application base image, I am forced to install the full set of gems when I first set up my development environment. Furthermore, if I haven't updated any gems in some feature branch, when the base image's gems are updated (daily, with trackler updates), I need to run `bundle install` again, rather than having the gem updates just come with the updated image. All that to say, there are some pretty clear wins in Docker land if we don't vendor gems in development.

Considering that, would you be opposed to my altering `bin/setup` not to use `--path=vendor/bundle`? With `vendor` and `.bundle` in `.gitignore`, the `vendor/bundle` directory and `.bundle/config` will remain for existing local dev environments, and things will continue to work as before, not requiring anyone to reinstall gems. New environments will instead install gems on the system—bundler's default behavior. I will add `vendor` and `.bundle` to `.dockerignore` (excludes these from the docker context and built images), so regardless of the presence of `vendor/`, Docker will exclude vendored gems and the bundler config.

@kytrinyx Are there any other caveats that I'm not thinking of?